CRM/Twingle/Form/Profile.php

```diff
@@ -139,61 +145,62 @@ public function buildQuickForm() {
       $this->_op = 'create';
     }
 
-    // Verify that a profile with the given name exists.
-    $profile_name = CRM_Utils_Request::retrieve('name', 'String', $this);
-    if (!$this->profile = CRM_Twingle_Profile::getProfile($profile_name)) {
-      $profile_name = NULL;
+    // Verify that a profile with the given id exists.
+
+    if ($this->_op != 'copy') {
+      $this->profile_id = CRM_Utils_Request::retrieve('id', 'Int', $this);
+      $this->profile = CRM_Twingle_Profile::getProfile($this->profile_id);
     }
 
     // Set redirect destination.
     $this->controller->_destination = CRM_Utils_System::url('civicrm/admin/settings/twingle/profiles', 'reset=1');
 
     switch ($this->_op) {
       case 'delete':
-        if ($profile_name) {
-          CRM_Utils_System::setTitle(E::ts('Delete Twingle API profile <em>%1</em>', array(1 => $profile_name)));
-          $this->addButtons(array(
-            array(
+        if ($this->profile_id) {
```

Comment:
This should check vor a valid profile instead, as `$profile_id` is basically just an integer retrieved from the request.

Reply:
It does not come only from the query parameters. At this stage it has already been validated in the `CRM_Twingle_Profile::getProfile($this->profile_id);` method (line 152) which returns NULL if the profile does not exist.

Reply:
Sorry, my fault. 😑
I will fix that.

Reply:
> PS. cross-posted with your edited answer.
Yes, sorry for the confusion. I only realized my mistake after sending the comment.